### PR TITLE
Closes #953 - Renamed AccessTokenRecord property from access_token_type to token_type

### LINKED ISSUE #953: [Cache] Rename AT property access_token_type to token_type
See [spec (corpnet required)](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/1757).

Summary:
- `AccessTokenRecord` property `access_token_type` is being renamed to `token_type`

Anticipated changes:
- Rename `@SerializedName()`
- Write custom deserializer to bridge backcompat

### PR DESCRIPTION
**Pull request by iambmelt**
Closes #953 

Renamed `AccessTokenRecord` property from `access_token_type` to `token_type`.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java [hunk 1]
```diff
@@ -86,7 +93,7 @@
      * utilize the access token to make a protected resource request (along with type-specific
      * attributes).
      */
-    @SerializedName(ACCESS_TOKEN_TYPE)
+    @SerializedName(value = TOKEN_TYPE, alternate = ACCESS_TOKEN_TYPE)
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java:96]: nice.

File: common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java [hunk 2]
```diff
@@ -37,16 +37,23 @@
...
 import static com.microsoft.identity.common.internal.dto.AccessTokenRecord.SerializedNames.TARGET;
+import static com.microsoft.identity.common.internal.dto.AccessTokenRecord.SerializedNames.TOKEN_TYPE;
 import static com.microsoft.identity.common.internal.dto.Credential.SerializedNames.EXPIRES_ON;
 
 public class AccessTokenRecord extends Credential {
 
     public static class SerializedNames extends Credential.SerializedNames {
         /**
-         * String of access token type.
+         * Deprecated string of access token type.
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java:47]:
```suggestion
         * <strong>Deprecated<strong> string of access token type.  Prefer @link{#TOKEN_TYPE} instead.
```

**Comment by AdamBJohnsonx** [comment]
Should we change the title of the pull request to reflect 'Renamed AccessTokenRecord property from access_token_type to token_type.' instead of 'Rename property access_token_type to access_token' ?

**Comment by iambmelt** [comment]
> Should we change the title of the pull request to reflect 'Renamed AccessTokenRecord property from access_token_type to token_type.' instead of 'Rename property access_token_type to access_token' ?

👍

File: common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java (hunk 1 above)
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java:96]: ❤

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java:96]: Isn't that slick?? I didn't know Gson could do that until.... now 😅

**Pull request merged**